--- origsrc/mate-text-editor-1.2.0/pluma/Makefile.am	2012-03-11 14:49:10.000000000 -0500
+++ src/mate-text-editor-1.2.0/pluma/Makefile.am	2012-04-17 16:02:17.300624900 -0500
@@ -226,9 +226,10 @@ install-exec-hook:
 if PLATFORM_WIN32
 	$(mkinstalldirs) "$(DESTDIR)$(libdir)"
 	$(INSTALL_DATA) libpluma.a "$(DESTDIR)$(libdir)"
-else
+endif
+if !OS_WIN32
 	rm -f $(DESTDIR)$(bindir)/mate-text-editor
-	ln -s pluma $(DESTDIR)$(bindir)/mate-text-editor
+	ln -s pluma$(EXEEXT) $(DESTDIR)$(bindir)/mate-text-editor
 endif
 
 if !OS_WIN32
